google/cloud/storage/testing/object_integration_test.cc

```diff
@@ -0,0 +1,86 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/testing/object_integration_test.h"
+#include "google/cloud/internal/getenv.h"
+#include "google/cloud/status.h"
+#include <gmock/gmock.h>
+#include <cstddef>
+#include <cstdlib>
+#ifdef __unix__
+#include <dirent.h>
+#endif  // __unix__
+#include <string>
+
+namespace google {
+namespace cloud {
+namespace storage {
+namespace testing {
+namespace {
+
+StatusOr<std::size_t> GetNumEntries(std::string const& path) {
+#ifdef __unix__
+  dirent** namelist;
+  int res = scandir(path.c_str(), &namelist, nullptr, alphasort);
+  if (res < 0) {
+    return Status(StatusCode::kInternal, "Failed to open directory \"" + path +
+                                             "\": " + strerror(errno));
+  }
+  for (int i = 0; i < res; ++i) {
+    free(namelist[i]);
+  }
+  free(namelist);
+  return res;
+#else   // __unix__
+  return Status(StatusCode::kUnimplemented,
+                "Can't check #entries in " + path +
+                    ", because only UNIX systems are supported");
+#endif  // __unix__
+}
+
+}  // anonymous namespace
+
+StatusOr<std::size_t> ObjectIntegrationTest::GetNumOpenFiles() {
+  auto res = GetNumEntries("/proc/self/fd");
+  if (!res) {
+    return res;
+  }
+  if (*res < 3) {
+    return Status(StatusCode::kInternal,
+                  "Expected at least three entries in /proc/self/fd: ., .., "
+                  "and the directory itself, found " +
+                      std::to_string(*res));
+  }
+  return *res - 3;
```

Comment:
I'm not sure why we bother with the `3` business given that we only deal in differentials, but it should probably just be `2` and forget about "the directory itself" bit.  It is, after all, an open file.